src/main/java/net/minecraftforge/common/extensions/IForgeItemStack.java

```diff
@@ -292,11 +292,21 @@ default float getXpRepairRatio()
     /**
      * Called to tick armor in the armor slot. Override to do something
      */
+    @Deprecated(forRemoval = true, since = "1.20.1") // Use onInventoryTick
     default void onArmorTick(Level level, Player player)
     {
         self().getItem().onArmorTick(self(), level, player);
     }
 
+    /**
+     * Called to tick this items in a players inventory, the indexes are the global slot index.
+     */
+    default void onInventoryTick(Level level, Player player, int slotIndex, int selectedIndex)
```

Comment:
This is not currently hooked up

Reply:
I'm not a fan of this existing at all, tbh - the signature is virtually the same to the vanilla one and it would be more realistic to just fix the index.  I realize we have to have it for bincompat, but yeah.

Reply:
Worse than that the way/spot it would be hooked into... would pass a different value for slotIndex than the implementation of this method expects so we would have to modify it decently before calling it.

Reply:
Since we have to keep this method for ABI, do we want to deprecate it, call it anyway, and just proxy to the vanilla method with the correct index? I need to scrub vanilla overrides and see if any of them actually call it, but the way the call currently is means the `slotIndex` is not actually usable, since you don't know which sub-inventory it came from.

Reply:
I am not sure how valid a solution that is as I [mentioned some on discord](https://discord.com/channels/313125603924639766/852298000042164244/1130154262421323847), given I definitely have used the "faulty" index for purposes of checking if something is on the hotbar or offhand but not in main inventory. Which the changes/decision with this would break (while the mod I do this on isn't on 1.20.1 yet, there may be other mods that do similar, and at the very least it would not be particularly evident that the way the index worked changed during porting)